### ISSUE TITLE
Error when using with AdvancedSlimeWorldManager

### ISSUE THREAD

**Issue by BingYanchi** [andrei1058/BedWars1058#528]
**Describe the bug**

I tried 1.17.1, 1.18, 1.18.1, all of which happened, which made it impossible to use ASWM
I've tried running ASWM and BedWars1058 separately without errors. But when they were used at the same time, the error appeared.

**To Reproduce**

1. put ASWM and BedWars1058 in plugin
2. put ASWM classmodifier
3. modify startup parameters

**Expected behavior**

Works fine with ASWM

**Screenshots**

![image](https://user-images.githubusercontent.com/36477925/170721485-ed5f1f9e-8a1a-483f-b40b-9d3d72d87b0f.png)

**System (please complete the following information):**
 - OS: Windows Server 2021
 - Server software paper
 - Server Version 1.18.1、1.18、1.17.1
 - Plugin Version 22.3.4

**Additional context**

**Comment by hugeBlack** [comment]
That's not an Advanced SWM issue, It's just because new version of Bedwars1058 doesn't included flowpower nbt. You can put its class in 1058's jar to solve this problem

**Comment by BingYanchi** [comment]
> 

how to do it

**Comment by BingYanchi** [comment]
> That's not an Advanced SWM issue, It's just because new version of Bedwars1058 doesn't included flowpower nbt. You can put its class in 1058's jar to solve this problem

how to do it

**Comment by hugeBlack** [comment]
下载 https://repo1.maven.org/maven2/com/flowpowered/flow-nbt/1.0.0/flow-nbt-1.0.0.jar
然后把jar包里的com文件夹复制到1058的jar包下面就行

然后这玩意1.18的bug挺多的，我已经修到能玩的程度了，你可以看看我的pr #509 和分支（

**Comment by BingYanchi** [comment]
> 下载 https://repo1.maven.org/maven2/com/flowpowered/flow-nbt/1.0.0/flow-nbt-1.0.0.jar 然后把jar包里的com文件夹复制到1058的jar包下面就行
> 
> 然后这玩意1.18的bug挺多的，我已经修到能玩的程度了，你可以看看我的pr #509 和分支（

我现在遇到一个很诡异的情况
![image](https://user-images.githubusercontent.com/36477925/170824296-17671094-b687-432a-abf4-130e0c170859.png)
虽然显示了六个竞技场可用，但是点开就是没有，用 /bw join 也是没有。控制台也没有报错。确认竞技场是启用的
如果你感兴趣可以到我的测试服务器看看，play.yistars.net
```
Java HotSpot(TM) 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
Starting org.bukkit.craftbukkit.Main
*** Warning, you've not updated in a while! ***
*** Please download a new build as per instructions from https://papermc.io/downloads ***
System Info: Java 17 (Java HotSpot(TM) 64-Bit Server VM 17.0.1+12-LTS-39) Host: Windows Server 2019 10.0 (amd64)
Loading libraries, please wait...
2022-05-28 19:49:13,998 ServerMain WARN Advanced terminal features are not available in this environment
[19:49:19 INFO]: Environment: authHost='https://authserver.mojang.com', accountsHost='https://api.mojang.com', sessionHost='https://sessionserver.mojang.com', servicesHost='https://api.minecraftservices.com', name='PROD'
[19:49:20 INFO]: Reloading ResourceManager: Default, bukkit
[19:49:21 INFO]: Loaded 7 recipes
[19:49:23 INFO]: Starting minecraft server version 1.18.1
[19:49:23 WARN]: ****************************
[19:49:23 WARN]: YOU ARE RUNNING THIS SERVER AS AN ADMINISTRATIVE OR ROOT USER. THIS IS NOT ADVISED.
[19:49:23 WARN]: YOU ARE OPENING YOURSELF UP TO POTENTIAL RISKS WHEN DOING THIS.
[19:49:23 WARN]: FOR MORE INFORMATION, SEE https://madelinemiller.dev/blog/root-minecraft-server/
[19:49:23 WARN]: ****************************
[19:49:23 INFO]: Loading properties
[19:49:23 INFO]: This server is running Paper version git-Paper-216 (MC: 1.18.1) (Implementing API version 1.18.1-R0.1-SNAPSHOT) (Git: 30cb7d0)
[19:49:23 INFO]: Server Ping Player Sample Count: 12
[19:49:23 INFO]: Using 4 threads for Netty based IO
[19:49:23 INFO]: Default game type: SURVIVAL
[19:49:23 INFO]: Generating keypair
[19:49:23 INFO]: Starting Minecraft server on *:22001
[19:49:24 INFO]: Using default channel type
[19:49:24 INFO]: Paper: Using Java 11 compression from Velocity.
[19:49:24 INFO]: Paper: Using Java cipher from Velocity.
[19:49:24 ERROR]: [STDERR] [org.bukkit.craftbukkit.v1_18_R1.legacy.CraftLegacy] Initializing Legacy Material Support. Unless you have legacy plugins and/or data this is a bug!
[19:49:29 WARN]: Legacy plugin BedWars1058-PopUpTowers v0.7 does not specify an api-version.
[19:49:29 WARN]: Legacy plugin BedWars1058-AntiDrop v1.2.1 does not specify an api-version.
[19:49:29 INFO]: [LuckPerms] Loading LuckPerms v5.4.25
[19:49:29 INFO]: [Vault] Loading Vault v1.7.3-b131
[19:49:29 INFO]: [eBackup] Loading eBackup v1.1.4
[19:49:29 INFO]: [SlimeWorldManager] Loading SlimeWorldManager v2.8.0
[19:49:30 INFO]: [PlaceholderAPI] Loading PlaceholderAPI v2.10.9
[19:49:30 INFO]: [BedWars1058] Loading BedWars1058 v22.3.4
[19:49:30 INFO]: [BedWars1058] Loading support for paper/spigot: v1_18_R1
[19:49:31 INFO]: [ProtocolLib] Loading ProtocolLib v4.8.0
[19:49:33 INFO]: [BedWars1058-PopUpTowers] Loading BedWars1058-PopUpTowers v0.7
[19:49:33 INFO]: [WorldEdit] Loading WorldEdit v7.2.10+1742f98
[19:49:33 INFO]: Got request to register class com.sk89q.worldedit.bukkit.BukkitServerInterface with WorldEdit [com.sk89q.worldedit.extension.platform.PlatformManager@11fb3d42]
[19:49:33 INFO]: [BedWars1058-Compass] Loading BedWars1058-Compass v1.4.2
[19:49:33 INFO]: [BedWars1058-AntiDrop] Loading BedWars1058-AntiDrop v1.2.1
[19:49:33 INFO]: Server permissions file permissions.yml is empty, ignoring it
[19:49:33 INFO]: [LuckPerms] Enabling LuckPerms v5.4.25
[19:49:34 INFO]:         __    
[19:49:34 INFO]:   |    |__)   LuckPerms v5.4.25
[19:49:34 INFO]:   |___ |      Running on Bukkit - Paper
[19:49:34 INFO]: 
[19:49:34 INFO]: [LuckPerms] Loading configuration...
[19:49:34 INFO]: [LuckPerms] Loading storage provider... [MYSQL]
[19:49:34 INFO]: [me.lucko.luckperms.lib.hikari.HikariDataSource] luckperms-hikari - Starting...
[19:49:34 INFO]: [me.lucko.luckperms.lib.hikari.HikariDataSource] luckperms-hikari - Start completed.
[19:49:35 INFO]: [LuckPerms] Loading messaging service... [SQL]
[19:49:35 INFO]: [LuckPerms] Loading internal permission managers...
[19:49:35 INFO]: [LuckPerms] Loading extension: me.lucko.luckperms.extension.legacyapi.LegacyApiExtension (extension-legacy-api-1.0.0.jar)
[19:49:35 INFO]: [LuckPerms] Performing initial data load...
[19:49:35 INFO]: [LuckPerms] Successfully enabled. (took 2011ms)
[19:49:35 INFO]: [Vault] Enabling Vault v1.7.3-b131
[19:49:35 INFO]: [Vault] [Permission] SuperPermissions loaded as backup permission system.
[19:49:35 INFO]: [Vault] Enabled Version 1.7.3-b131
[19:49:36 INFO]: [LuckPerms] Registered Vault permission & chat hook.
[19:49:36 INFO]: [BedWars1058] Enabling BedWars1058 v22.3.4
[19:49:36 INFO]: [BedWars1058] Hook into SlimeWorldManager support!
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g0 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Loading arena: bw_temp_y22m05d28g0
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g1 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g2 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g3 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g4 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g5 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g6 was added to the enable queue.
[19:49:36 INFO]: [BedWars1058] Arena bw_temp_y22m05d28g7 was added to the enable queue.
[19:49:36 ERROR]: [BedWars1058] [STDERR] [Server thread] INFO com.andrei1058.bedwars.libs.hikari.HikariDataSource - BedWars1058MySQLPool - Starting...
[19:49:36 WARN]: Nag author(s): '[andrei1058]' of 'BedWars1058' about their usage of System.out/err.print. Please use your plugin's logger instead (JavaPlugin#getLogger).
[19:49:36 ERROR]: [BedWars1058] [STDERR] [Server thread] INFO com.andrei1058.bedwars.libs.hikari.HikariDataSource - BedWars1058MySQLPool - Start completed.
[19:49:36 INFO]: [BedWars1058] Hooked into PlaceholderAPI support!
[19:49:36 INFO]: [PlaceholderAPI] Successfully registered expansion: bw1058
[19:49:37 INFO]: [ProtocolLib] Enabling ProtocolLib v4.8.0
[19:49:37 INFO]: [ProtocolLib] Started structure compiler thread.
[19:49:37 INFO]: [WorldEdit] Enabling WorldEdit v7.2.10+1742f98
[19:49:37 INFO]: Registering commands with com.sk89q.worldedit.bukkit.BukkitServerInterface
[19:49:37 INFO]: WEPIF: Vault detected! Using Vault for permissions
[19:49:37 INFO]: Using com.sk89q.worldedit.bukkit.adapter.impl.v1_18_R1.PaperweightAdapter as the Bukkit adapter
[19:49:38 WARN]: **** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!
[19:49:38 WARN]: The server will make no attempt to authenticate usernames. Beware.
[19:49:38 WARN]: Whilst this makes it possible to use BungeeCord, unless access to your server is properly restricted, it also opens up the ability for hackers to connect with any username they choose.
[19:49:38 WARN]: Please see http://www.spigotmc.org/wiki/firewall-guide/ for further information.
[19:49:38 WARN]: To change this, set "online-mode" to "true" in the server.properties file.
[19:49:38 INFO]: Preparing level "world"
[19:49:40 INFO]: Preparing start region for dimension minecraft:overworld
[19:49:40 INFO]: Time elapsed: 353 ms
[19:49:40 INFO]: [eBackup] Enabling eBackup v1.1.4
[19:49:40 INFO]: [eBackup] Initializing eBackup...
[19:49:40 INFO]: [eBackup] Configured the cron task to be: at 02:00
[19:49:40 INFO]: [eBackup] Plugin initialized!
[19:49:40 INFO]: [SlimeWorldManager] Enabling SlimeWorldManager v2.8.0
[19:49:40 INFO]: [PlaceholderAPI] Enabling PlaceholderAPI v2.10.9
[19:49:40 INFO]: [PlaceholderAPI] Fetching available expansion information...
[19:49:40 INFO]: [BedWars1058-PopUpTowers] Enabling BedWars1058-PopUpTowers v0.7*
[19:49:40 INFO]:  
[19:49:40 INFO]: BedWars1058 Pop-Up Towers
[19:49:40 INFO]: by kimoVoid
[19:49:40 INFO]:  
[19:49:40 INFO]: [BedWars1058-Compass] Enabling BedWars1058-Compass v1.4.2
[19:49:40 INFO]: [BedWars1058-Compass] Hook into Vault chat support
[19:49:40 INFO]: [BedWars1058-Compass] Successfully loaded in 274ms
[19:49:40 INFO]: [BedWars1058-AntiDrop] Enabling BedWars1058-AntiDrop v1.2.1*
[19:49:41 INFO]: [PlaceholderAPI] Placeholder expansion registration initializing...
[19:49:41 INFO]: Running delayed init tasks
[19:49:41 INFO]: [BedWars1058] Hooked into vault chat support!
[19:49:41 INFO]: [BedWars1058] Vault found, but no economy provider!
[19:49:41 INFO]: No expansions were registered!
[19:49:41 INFO]: Done (18.075s)! For help, type "help"
[19:49:41 INFO]: Timings Reset
[19:49:41 INFO]: [SWM] Loading world airshow.
[19:49:41 INFO]: [SWM] World airshow loaded in 257ms.
[19:49:41 INFO]: [BedWars1058] Loading internal Party system. /party
[19:49:46 INFO]: [BedWars1058] This server is running in BUNGEE with auto-scale true
[19:49:58 INFO]: UUID of player Bing_Yanchi is 186dfa0d-8e66-4302-80a9-110bd79464b5
[19:49:58 INFO]: Bing_Yanchi[/117.88.162.18:1241] logged in with entity id 76 at ([world]0.0, 65.0, 0.0)
[19:50:03 INFO]: Bing_Yanchi issued server command: /bw arenaList
[19:50:10 INFO]: Bing_Yanchi issued server command: /bw build
[19:50:13 INFO]: Bing_Yanchi issued server command: /bw build
[19:50:14 INFO]: Bing_Yanchi issued server command: /bw compass
[19:50:16 INFO]: Bing_Yanchi issued server command: /bw compass menu
[19:50:17 INFO]: Bing_Yanchi issued server command: /bw compass menu
[19:50:19 INFO]: Bing_Yanchi issued server command: /bw compass reload
[19:51:05 INFO]: Bing_Yanchi issued server command: /bw arenaList
[19:51:08 INFO]: Bing_Yanchi issued server command: /bw
[19:51:10 INFO]: Bing_Yanchi issued server command: /bw arenaList
```

**Comment by hugeBlack** [comment]
enable了吗

**Comment by BingYanchi** [comment]
> enable了吗

enable 了
![image](https://user-images.githubusercontent.com/36477925/170826034-261ed540-65f1-42e9-9a14-3fa5162bd592.png)

**Comment by hugeBlack** [comment]
看了眼日志，swm没有去加载竞技场所在的世界
enable的时候会去调getRestoreAdapter的onEnable(IArena)方法，然而22.3.4在当RestoreAdapter是swm的时候onEnable是有bug的（

你可以用我的分支来构建一下，看看行不行（

**Comment by BingYanchi** [comment]
> 看了眼日志，swm没有去加载竞技场所在的世界 enable的时候会去调getRestoreAdapter的onEnable(IArena)方法，然而22.3.4在当RestoreAdapter是swm的时候onEnable是有bug的（
> 
> 你可以用我的分支来构建一下，看看行不行（

有分支的构建么

**Comment by hugeBlack** [comment]
https://hugeblack.lanzoul.com/iGzzU05kmvcb

**Comment by BingYanchi** [comment]
> https://hugeblack.lanzoul.com/iGzzU05kmvcb

似乎可以了，感谢！thanks！

**Comment by andrei1058** [comment]
I see you keep talking in chinese through issues, I can't help that way..

**Comment by tomkeuper** [comment]
Closed in favour of #624
